### PR TITLE
MULE-13711: JMS Publisher always joins the current TX, if one exists,…

### PR DESCRIPTION
**Pull request by estebanwasinger**
… regardless the chosen transactional Action

### CODE REVIEW

File: modules/extensions-spring-support/src/test/resources/models/tx-ext.json [hunk 1]
```diff
@@ -427,15 +427,15 @@
...
-                      "sequence": 3
-                    },
                     "org.mule.runtime.extension.api.property.QNameModelProperty": {
                       "value": {
                         "namespaceURI": "http://www.mulesoft.org/schema/mule/core",
                         "localPart": "abstract-reconnection-strategy",
                         "prefix": "mule"
                       }
+                    },
+                    "org.mule.runtime.extension.api.property.InfrastructureParameterModelProperty": {
```
Comment (marianogonzalez) [inline modules/extensions-spring-support/src/test/resources/models/tx-ext.json]: is this correct or just pablon's bug?

File: modules/extensions-spring-support/src/test/resources/transactional-parameters-injection-config.xml [hunk 2]
```diff
@@ -24,7 +24,7 @@
     </flow>
 
     <flow name="injectInOperationJoinNotSupported">
-        <tx:inject-transactional-action config-ref="localTx1" action="NOT_SUPPORTED"/>
+        <tx:inject-transactional-action config-ref="localTx1" transactionalAction="NOT_SUPPORTED"/>
```
Comment (marianogonzalez) [inline modules/extensions-spring-support/src/test/resources/transactional-parameters-injection-config.xml]: why does this changes name? Wasn't transactionalAction it's actual name?

Comment (estebanwasinger) [reply modules/extensions-spring-support/src/test/resources/transactional-parameters-injection-config.xml]: the name must be always `transactionalAction` the bug I fixed now fixes the name to be always `transactionalAction` doesn't matter how is defined in the Java Code

File: modules/extensions-spring-support/src/test/resources/models/tx-ext.json (hunk 1 above)
Comment (estebanwasinger) [reply modules/extensions-spring-support/src/test/resources/models/tx-ext.json]: pablon's bug

**Pull request merged**